### ISSUE TITLE
Please bump transformers bounds

### ISSUE THREAD

**Issue by pjonsson** [mainland/exception-transformers#3]
I think this is the right package to report the bug on.

We're using language-c-quote to feed strange programs into feldspar-compiler for debugging purposes. We're also using QuickCheck for the test suite and there seems to be some funny interaction between transformers and exception-transformers on older versions of GHC. Here's a snippet of the build log (from https://travis-ci.org/Feldspar/feldspar-compiler/jobs/114312170):

```
..
trying: feldspar-compiler-0.6.1.0:*test
trying: QuickCheck-2.8.2 (dependency of feldspar-compiler-0.6.1.0:*test)
trying: transformers-0.5.2.0 (dependency of QuickCheck-2.8.2)
trying: tasty-golden-2.3.1 (dependency of feldspar-compiler-0.6.1.0:*test)
trying: async-2.1.0 (dependency of tasty-golden-2.3.1)
trying: stm-2.4.4.1 (dependency of async-2.1.0)
trying: language-c-quote-0.11.4.1 (dependency of feldspar-compiler-0.6.1.0)
next goal: exception-transformers (dependency of language-c-quote-0.11.4.1)
rejecting: exception-transformers-0.4.0.3, 0.4.0.2, 0.4.0.1, 0.4, 0.3.0.4
(conflict: transformers==0.5.2.0, exception-transformers => transformers>=0.2
&& <0.5)
rejecting: exception-transformers-0.3.0.3 (conflict: transformers==0.5.2.0,
exception-transformers => transformers>=0.2 && <0.4)
rejecting: exception-transformers-0.3.0.2 (conflict: stm==2.4.4.1,
exception-transformers => stm>=2.1 && <2.4)
rejecting: exception-transformers-0.3.0.1, 0.3 (conflict: stm==2.4.4.1,
exception-transformers => stm>=2.1 && <2.3)
rejecting: exception-transformers-0.2 (conflict: language-c-quote =>
exception-transformers>=0.3 && <0.5)
Backjump limit reached (change with --max-backjumps).
```

I don't have an easy way to test older versions of GHC but with 7.8.3 I can build exception-transformers if I just bump the upper bound on transformers.

**Comment by mainland** [comment]
The transformers package is bundled with GHC---GHC 7.8.3 bundles transformers-0.3.0.0. I don't understand how you are running into a version bounds error on GHC 7.8.3.

**Comment by pjonsson** [comment]
Oh, my apologies. The error log in question is for GHC 7.6.3.

I have considered removing the language-c-quote dependency from feldspar-compiler build but right now it is a bit too useful for us so I'd prefer to find a way to make it work.

**Comment by mainland** [comment]
GHC 7.6.3 ships with transformers-0.3.0.0 as well. Are you sure you're not using a more recent GHC?

I am very confused as to why you would be getting this error with GHC 7.6 or 7.8. What package is requiring transformers >= 0.5?

**Comment by pjonsson** [comment]
Line 308 in the linked log reports the GHC version. Line 304 reports the cabal version. I'm not sure what the odds are that Travis is inconsistent with the environment during the build of the package but I'm guessing it's a fairly small risk.

My interpretation of the log is that QuickCheck 2.8.2 is the package which pulls in transformers 0.5. The constraint in QuickCheck is only transformers >= 0.2 but we get 0.5 since that is available.

**Comment by pjonsson** [comment]
Didn't think about asking but I'll ask now: does the linked log require a login or can you look at it?

**Comment by mainland** [comment]
Seems to work fine on 7.8 and 7.10, which is odd. Perhaps it's time for you to drop support for 7.6 ;)

I was planning on bumping version bounds after 8.0 is released. I hope you can wait until then.

**Comment by mainland** [comment]
Fixed in aaa7df16.
